src/main/kotlin/org/rust/lang/core/psi/ext/RsFunction.kt

```diff
@@ -121,7 +122,7 @@ abstract class RsFunctionImplMixin : RsStubbedNamedElementImpl<RsFunctionStub>,
     override val crateRelativePath: String? get() = RsPsiImplUtil.crateRelativePath(this)
 
     final override val innerAttrList: List<RsInnerAttr>
-        get() = block?.innerAttrList.orEmpty()
+        get() = PsiTreeUtil.getStubChildrenOfTypeAsList(this, RsInnerAttr::class.java)
```

Comment:
This change is actually wrong.
The RsInnerAttr are children of the RsBlock.
But the new code searches them inside the RsFunction

Reply:
Fixed, i also added tests for inner/outer attribute owners and some additional fixes for problems i found.